src/pages/MyPage/History/History.js

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useEffect } from 'react';
+import HistoryTable from './HistoryTable';
+import HistoryHeader from './HistoryHeader';
+import { isValidObject } from '../../../utils';
+import styled from 'styled-components';
+
+function History() {
+  const [data, setData] = useState({});
+  const [search, setSearch] = useState('');
+  const [filterValue, setFilterValue] = useState('');
+
+  useEffect(() => {
+    fetch('/data/history.json')
+      .then(res => res.json())
+      .then(res => setData(res));
+  }, []);
+
+  const filterHandler = e => {
+    setFilterValue(e.target.value);
+  };
+
+  const handleChange = e => {
+    setSearch(e.target.value);
+  };
+
+  const filteredList =
+    isValidObject(data) &&
+    data.results.filter(product => {
+      return (
+        product.item.includes(search) &&
+        (filterValue !== '전체'
+          ? product.status === filterValue
+          : product.status)
+      );
+    });
+
+  return (
+    <>
+      {isValidObject(data) && (
+        <HistoryWrapper>
+          <div>
+            <SearchBar onChange={handleChange}>
+              <i class="fas fa-search"></i>
+              <input
+                type="text"
+                placeholder="상품호수 또는 상품명을 검색해보세요"
+              />
+            </SearchBar>
+            <HistoryMain>
+              <HistoryHeader filterHandler={filterHandler} />
+              <PaymentSummary>
+                <li>
+                  <div>
+                    <h3>투자금액</h3>
+                    <span>
+                      {filteredList
+                        .reduce((sum, item) => (sum += item.invest), 0)
+                        .toLocaleString()}
```

Comment:
누적값 구하는 부분 누적값 구해야되는 프로퍼티만 바뀌고 여러군데서 재활용되고 있는 것 같은데 하나의 함수로 재사용 가능하지 않을까요?

```
const getReducedValueOf = (property) => arr.reduce((acc,cur)=>{
  return acc + cur[property];
},0).toLocaleString();
```

```suggestion
                      {getReducedValueOf("invest")}
```

Reply:
오호... 반영하겠습니다!